packages/notifi-react/lib/context/NotifiTargetContext.tsx

```diff
@@ -310,16 +316,74 @@ export const NotifiTargetContextProvider: FC<PropsWithChildren> = ({
       .filter((item): item is Target => !!item);
   }, [targetInfoPrompts, targetData]);
 
+  const EMAIL_REGEX = /^[a-zA-Z0-9._:$!%-+]+@[a-zA-Z0-9.-]+.[a-zA-Z]$/;
+  const TELEGRAM_REGEX =
+    /^@?(?=\w{5,32}\b)[a-zA-Z0-9]+(?:[a-zA-Z0-9_ ]+[a-zA-Z0-9])*$/;
+
+  const validateEmail = (
+    email: string,
+  ): { isValid: boolean; error?: string } => {
+    if (email === '') {
+      return { isValid: true };
+    }
+    const isValid = EMAIL_REGEX.test(email);
+    return {
+      isValid,
+      error: isValid ? undefined : 'Invalid email address',
+    };
+  };
+
+  const validateTelegram = (
+    telegramId: string,
+  ): { isValid: boolean; error?: string } => {
+    if (telegramId === '') {
+      return { isValid: true };
+    }
+    const isValid = telegramId.length > 1 && TELEGRAM_REGEX.test(telegramId);
+    return {
+      isValid,
+      error: isValid ? undefined : 'Invalid telegram ID',
+    };
+  };
+
   const updateTargetInputs = useCallback(
     <T extends 'form' | 'toggle'>(
       target: T extends 'form' ? FormTarget : ToggleTarget,
-      value: T extends 'form' ? { value: string; error?: string } : boolean,
+      value: T extends 'form' ? TargetInputFromValue : boolean,
```

Comment:
This is also included in the incoming stage branch which will be merged into main next Tuesday.